ci: CircleCI builds are failing since a recent commit to the master

```
Removing intermediate container 9c2d63246ff7
Step 11/13 : WORKDIR $EKSCTL_BUILD/vendor/github.com/goreleaser/goreleaser
 ---> a740dc3e3269
Removing intermediate container 98851059cc6b
Step 12/13 : RUN make build && go install
 ---> Running in 342ef9f889b4
go build
go: finding github.com/Masterminds/semver v1.4.2
go: finding github.com/mattn/go-isatty v0.0.4
...
*snip*
...
go: downloading github.com/jmespath/go-jmespath v0.0.0-20160202185014-0b12d6b521d8
# net
/usr/local/go/src/net/cgo_linux.go:10:19: fatal error: netdb.h: No such file or directory
 #include <netdb.h>
                   ^
compilation terminated.
make: *** [Makefile:44: build] Error 2
The command '/bin/sh -c make build && go install' returned a non-zero code: 2
make: *** [eksctl-build-image] Error 2
Exited with code 2
```

![image](https://user-images.githubusercontent.com/22009/50427964-ccf30880-08f5-11e9-9880-0d6db57951fe.png)

I've reproduced and verified that musl-dev fixes the issue, thanks @mukaibot :)

```
$ docker run -it golang:1.11-alpine3.8
# apk add --update bash gcc git make
# bash
```

One thing worth noting is that, `go get` works without complaining about the cgo-related error. Why?

```
# This works! But why? Where does the CGO dependency go when we build it via `go get`?
$ go get github.com/goreleaser/goreleaser

$ cd /go/src/github.com/goreleaser/goreleaser
```

`make build` breaks, as we might have expected:

```
$ make build
go build
# net
/usr/local/go/src/net/cgo_linux.go:10:19: fatal error: netdb.h: No such file or directory
 #include <netdb.h>
                   ^
compilation terminated.
make: *** [Makefile:44: build] Error 2
```

Installing musl-dev does fix it:

```
$ apk add musl-dev
(1/1) Installing musl-dev (1.1.19-r10)
OK: 122 MiB in 38 packages
$ make build
go build
$ echo $?
0
```

--- a/build/Dockerfile
+++ b/build/Dockerfile
@@ -24,6 +24,6 @@ RUN dep ensure
 RUN ./install.sh
 
 WORKDIR $EKSCTL_BUILD/vendor/github.com/goreleaser/goreleaser
-RUN make build && go install
+RUN CGO_ENABLED=0 make build && go install
 
 WORKDIR $GOPATH

